src/app/components/moj/moj-sortable-table/readme.md

```diff
@@ -0,0 +1,228 @@
+
+```html
+
+<app-moj-sortable-table>
+  <ng-container head>
+    <th
+      app-moj-sortable-table-header
+      columnKey="imposition"
+      [sortDirection]="sortState['imposition']"
+      (sortChange)="onSortChange($event)"
+    >
+      Imposition
+    </th>
+    <th
+      app-moj-sortable-table-header
+      columnKey="creditor"
+      [sortDirection]="sortState['creditor']"
+      (sortChange)="onSortChange($event)"
+    >
+      Creditor
+    </th>
+    <th
+      app-moj-sortable-table-header
+      columnKey="amountImposed"
+      [sortDirection]="sortState['amountImposed']"
+      (sortChange)="onSortChange($event)"
+    >
+      Amount imposed
+    </th>
+    <th
+      app-moj-sortable-table-header
+      columnKey="amountPaid"
+      [sortDirection]="sortState['amountPaid']"
+      (sortChange)="onSortChange($event)"
+    >
+      Amount paid
+    </th>
+    <th
+      app-moj-sortable-table-header
+      columnKey="balanceRemaining"
+      [sortDirection]="sortState['balanceRemaining']"
+      (sortChange)="onSortChange($event)"
+    >
+      Balance remaining
+    </th>
+  </ng-container>
+  <ng-container row>
+    @for (row of abstractTableData; track row) {
+      <tr app-moj-sortable-table-row>
+        <td app-moj-sortable-table-row-data id="imposition">{{ row['imposition'] }}</td>
+        <td app-moj-sortable-table-row-data id="creditor">
+          {{
+            row['creditor'] === 'major'
+              ? 'Major Creditor'
+              : row['creditor'] === 'minor'
+                ? 'Minor Creditor'
+                : 'Default Creditor'
+          }}
+        </td>
+        <td app-moj-sortable-table-row-data id="amountImposed">{{ row['amountImposed'] }}</td>
+        <td app-moj-sortable-table-row-data id="amountPaid">{{ row['amountPaid'] }}</td>
+        <td app-moj-sortable-table-row-data id="balanceRemaining">{{ row['balanceRemaining'] }}</td>
+      </tr>
+    }
+  </ng-container>
+</app-moj-sortable-table>
+
+
+```
+
+These are some of the interfaces you will need to create for your component.
+```typescript
+    import { ISortStateInterface } from '@components/abstract/abstract-sortable-table/interfaces/abstract-sortable-table-interfaces';
+    import { IObjectSortableInterface } from '@services/sort-service/interfaces/sort-service-interface';
+
+    export interface ITableComponentTableData extends IObjectSortableInterface<string | number> {
+    imposition: string;
+    creditor: string;
+    amountImposed: number;
+    amountPaid: number;
+    balanceRemaining: number;
+    }
+
+    export interface ISortState extends ISortStateInterface {
+    imposition: 'ascending' | 'descending' | 'none';
+    creditor: 'ascending' | 'descending' | 'none';
+    amountImposed: 'ascending' | 'descending' | 'none';
+    amountPaid: 'ascending' | 'descending' | 'none';
+    balanceRemaining: 'ascending' | 'descending' | 'none';
+    }
+```
```

Comment:
Should this whole document be updated to reflect the changes made. For example IObjectSortableInterface has been renamed has it not?